webui/react/src/pages/TrialLogs.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { useParams } from 'react-router';
+
+import LogViewer, { LogViewerHandles } from 'components/LogViewer';
+import Page from 'components/Page';
+import UI from 'contexts/UI';
+import usePolling from 'hooks/usePolling';
+import { useRestApiSimple } from 'hooks/useRestApi';
+import { getTrialLogs } from 'services/api';
+import { TrialLogsParams } from 'services/types';
+import { Log } from 'types';
+
+interface Params {
+  trialId: string;
+}
+
+const TAIL_SIZE = 10000;
+
+const TrialLogs: React.FC = () => {
```

Comment:
question: the logic and this component looks a lot like the master logs code. did you want to explicitly generalize it in another ticket/PR?